### PR TITLE
Add other opening explorer endpoints

### PR DESCRIPTION
**Pull request by rpesche**
This PR add thow opening explore endpoints : ``/masters`` and ``/player``.
They allow querying masters games and specific player games database respectively.

The ``/player`` endpoint doesn't always return curated data as they may not be available at the time.
Instead, it stream partial results as they are computed.

I added a ``wait`` parameter to specify which result we wan't: first available, last result or result after a certain amount of time.

### CODE REVIEW

**Comment by benediktwerner** [comment]
Thanks!

The streaming endpoint seems a bit tricky, having the option to stream it vs waiting seems reasonable, specifying a timeout seems a bit misleading though since it only times out on the next message from Lichess, even if that is maybe a minute later. And it's probably not that common that you only want partial results? So I'd rather let users implement behavior like this themselves if they want to.

Also, seems like a good idea to add some documentation for the function params and a return type for the last method.

**Comment by rpesche** [comment]
I agree about the partial result: no application could deal with they and they must be ignored.

On the other hand, one could want its application to not "stuck" on this operation, which can take a long time (several minutes for some player). The timeout is a feature to forget about this operation, as it take to much time, and, maybe, get the final result after.

Thinking of that, maybe I should return another parameter indicating that it's partial result. Something like this:


```python
for response in self._r.get(path, params=params, stream=True):
    step = time.time() - start
                                                               
    # We don't wan't to wait, and return the first result
    if wait is False:
        break
                                                               
    # The expected time is over, we return the current result
    if not isinstance(wait, bool) and step > wait:
        yield cast(OpeningStatistic, response), False
                                                               
return cast(OpeningStatistic, response), True
```

Or maybe put another ``partial: bool`` in a new ``PartialOpeningStatistic(OpeningStatistic)``

**Comment by benediktwerner** [comment]
Hm, yeah, that seems like a good idea. Either option seems fine, using a second return value makes it a bit more obvious but putting it into the response object seems a bit cleaner and is nicer if you don't need to check it bc you wait anyways.

Actually, I think there also should be a way to wait without getting partial results, since they usually won't be needed.

Regarding the timeout: The thing that I don't like about it is that it can only time out after the next Lichess response. That's rather confusing and given that it's a rather niche feature and users can easily enough implement this themselves, I'd rather not do this. Though I guess we could pass the timeout to requests instead which will then actually time out immediately when the time is up. And that's also not something users can implement themselves that easily.

File: tests/clients/test_opening_explorer.py [hunk 1]
```diff
@@ -34,3 +34,44 @@ def test_correct_rating_params(self):
...
+class TestPlayerGames:
+    @pytest.mark.vcr
+    @pytest.mark.default_cassette("TestPlayerGames.results.yaml")
+    def test_wait_for_last_results(self):
+        result = Client().opening_explorer.get_player_games(
+            player="evachesss", color="white", wait=True
+        )
+        assert result["white"] == 125
+        assert result["draws"] == 18
+        assert result["black"] == 133
```
Comment (kraktus) [inline tests/clients/test_opening_explorer.py:59]: Hey, is there any particular reason to check for the exact number of games? This make the test brittle, and I don't think it tests anything more than just sending the requests.

Comment (benediktwerner) [reply tests/clients/test_opening_explorer.py:59]: It tests that the response is correct, e.g. really waits until the indexing is finished. It's true that hard-coding numbers mean you need to adjust them if you record new response mocks but that shouldn't really be necessary so seems fine imo.

**Pull request merged**